user-contrib/Ltac2/tac2entries.ml

```diff
@@ -869,6 +869,32 @@ end
 
 (** Printing *)
 
+let print_ltac_type qid =
+  if Tac2env.is_constructor qid then
+    let mkn =
+      try Some (Tac2env.locate_constructor qid)
+      with Not_found -> None
+    in
+    match mkn with
+    | Some kn ->
+      let data = Tac2env.interp_constructor kn in
+      let t = data.Tac2env.cdata_type in
+      Feedback.msg_notice (hov 2 (pr_qualid qid ++ spc () ++ str ":" ++ spc () ++ pr_typref t))
+    | None -> Feedback.msg_notice (hov 2 (pr_qualid qid ++ spc () ++ str ":" ++ spc () ++ str "Type"))
```

Comment:
This should probably be an error.

Reply:
This gets triggered by `Check Ltac2 ABC` where `ABC` is a type.

Reply:
Oh, I didn't realize that uppercase types were allowed. This looks more like a bug than anything else to me :/ In any case, this shouldn't read "XYZ : Type" as there is no such thing as `Type` in the Ltac2 type system.